publish metrics for duration between form creation time and pega upda…

brostk opened the pull request:
…te call

Keep your PR as a Draft until it's ready for Platform review. A PR is ready for Platform review when it has a teammate approval and tests, linting, and settings checks pass CI. See [these tips](https://depo-platform-documentation.scrollhelp.site/developer-docs/vets-api-pr-tips) on how to avoid common delays in getting your PR merged.

## Summary

- *This work is behind a feature toggle (flipper): YES*
- Measure the duration between the form creation time from the `submit` call and the Pega callback and publish metrics.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/117775

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

brostk commented:
> Can we do this without disabling the rubocop Metrics/MethodLength rule? Otherwise everything looks good.

Updated 👍

stevelong00 reviewed (approved):
Looks great!

RachalCassity commented on modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb:
```diff
@@ -135,6 +145,42 @@ def fetch_forms_by_uuid(form_uuid)
+        return unless form_files.any?
+
+        # Calculate and publish metrics for first matching form file
+        form_files.first do |form|
```
Is form_files.first an array? Or should it be `if (form = form_files.first)`

RachalCassity replied:
Is form_files.first an array? Or should it be if (form = form_files.first). Please add the new method to the corresponding spec file.

RachalCassity reviewed (changes requested):
Add method to spec files

brostk commented:
> Can we do this without disabling the rubocop Metrics/MethodLength rule? Otherwise everything looks good.

Updated 👍 

> Add method to spec files

@RachalCassity Added 👍  And simplified code for your other comment as well.
